src/eos-updater-poll.c

```diff
@@ -240,14 +428,49 @@ metadata_fetch (GTask *task,
       return;
     }
 
-  get_fetchers (&config, &fetchers, &source_variants);
-  info = run_fetchers (fetch_data,
-                       fetchers,
-                       config.download_order);
+  /* Do we want to use the new libostree code for P2P, or fall back on the old
+   * eos-updater code?
+   * FIXME: Eventually drop the old code. See:
+   * https://phabricator.endlessm.com/T19606 */
+  if (use_new_code)
+    {
+      info = metadata_fetch_new (data->repo, &config, task_context, cancellable, &error);
+
+      if (error != NULL)
```

Comment:
should we check `info == NULL` as opposed to `error != NULL` ?

Reply:
Nope, since it’s permitted for `metadata_fetch_new()` to return `NULL` and not set an error, in the case that no results were found.